tmereader/repository.go

```diff
@@ -80,7 +98,8 @@ func (t *tmeRepository) GetTmeTermsFromIndex(startRecord int) ([]interface{}, er
 }
 
 func (t *tmeRepository) getTmeTermsInChunks(startPosition int, maxRecords int) ([]interface{}, error) {
-	req, err := http.NewRequest("GET", fmt.Sprintf("%s/rs/authorityfiles/%s/terms?maximumRecords=%d&startRecord=%d", t.tmeBaseURL, t.taxonomyName, maxRecords, startPosition), nil)
+	url := fmt.Sprintf("%s/rs/%s/%s/%s?maximumRecords=%d&startRecord=%d", t.tmeBaseURL, t.source.String(), t.taxonomyName, sourcePathSuffix[t.source], maxRecords, startPosition)
```

Comment:
If you take the source through a String method, i would do something similar for the path suffix too (PathSuffix()). Or we can take the string simply from the array, without a function, keeping both of the references consistent.